plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/TrinoToMySqlWriteChecker.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mysql;
+
+import io.trino.spi.TrinoException;
+
+import java.time.Instant;
+
+import static io.trino.spi.StandardErrorCode.INVALID_ARGUMENTS;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TrinoToMySqlWriteChecker<T>
```

Comment:
I don't think I understand the complexity introduced with these abstractions. The only two production usages of the validate() method I can see is that we pass an Instance and the code checks under the covers if it's within epoch's min and max values. Do we really need to introduce private interface, private class, another private class for Range in order to do that?

It seems that the epoch min and epoch max are hardcoded anyway. Isn't it the case that the only thing we need is a

```
private static boolean validate(Instant i) {
   return i >= Instant.parse("1970-01-01T00:00:00.000000Z") && Instant.parse("2038-01-19T03:14:07.499999Z")
}
```
(or something like above that compiles or throws an exception instead of returning a boolean) in MySqlClient class?

I would understand wrapping into classes, if the code was unit-tested. But it does not look like there's a unit test for this class?

Reply:
Agreed this could be simplified into a static `validateMySqlTimestampValue` function in `MySqlClient`.